### ISSUE TITLE
`functional/prefer-readonly-type` false positive on tuple types

### ISSUE THREAD

**Issue by maxxxxxdlp** [jonaskello/eslint-plugin-functional#385]
# Bug Report

Given the following code, ESLint reports `functional/prefer-readonly-type` rule:

```ts
type A = Readonly<[string]>;
```

## Expected behavior

No error should be reported as the non-readonly tuple is never actually created since it is immediately being cast into a readonly tuple.

## Actual behavior

Autofix replaces the above with:

```ts
type A = Readonly<readonly [string]>;
```

## Proposed changes

Don't report the rule in this case, or disable auto fix, or remove `Readonly<>` when adding `readonly `

**Comment by maxxxxxdlp** [comment]
For reference, the `Readonly<>` syntax is preferred because it can be used for the following:

```ts
export interface Conformation extends Readonly<[number, ...Conformations]> {}
```

Where as equivalent using `readonly []` is a syntax error:

```ts
export interface Conformation extends readonly [number, ...Conformations] {}
```

**Comment by RebeccaStevens** [comment]
We currently don't have support for `Readonly` (see #51).

I've been working on the next release which will add support for this.
